picoCTF-web/api/logger.py

```diff
@@ -140,29 +139,47 @@ def setup_logs(args):
     stats_log.setLevel(logging.INFO)
     log.root.addHandler(stats_log)
 
+def _remove_parameter(arg_dict, param_path):
+    """Recurses through a dictionary of dictionaries and removes the given param"""
+    if param_path[0] not in arg_dict:
+        return arg_dict
 
-def log_action(f):
-    """Log a function invocation and its result."""
 
-    @wraps(f)
-    def wrapper(*args, **kwargs):
-        """Provide contextual information to the logger."""
-        log_information = {
-            "name": "{}.{}".format(f.__module__, f.__name__),
-            "args": args,
-            "kwargs": kwargs,
-            "result": None,
-        }
-        try:
-            log_information["result"] = f(*args, **kwargs)
-        except Exception as error:
-            log_information["exception"] = error
-            raise error
-        finally:
-            log.info(log_information)
-        return log_information["result"]
-
-    return wrapper
+    new_dict = arg_dict.copy() # Needed to avoid aliasing effects
+    if len(param_path) == 1:
+        new_dict[param_path[0]] = "REDACTED"
+        return new_dict
+
+    new_dict[param_path[0]] = _remove_parameter(arg_dict[param_path[0]], param_path[1:])
+    return new_dict
+
+def log_action(dont_log=[]):
+    """Log a function invocation and its result."""
+    def outer_wrapper(f):
+        @wraps(f)
+        def wrapper(*args, **kwargs):
+            """Provide contextual information to the logger."""
+
+            func_args = inspect.getcallargs(f, *args, **kwargs)
```

Comment:
`inspect.getcallargs` is [deprecated](https://docs.python.org/3.7/library/inspect.html#inspect.getcallargs), so to be more futureproof we can get the same behavior with:
```python
func_sig = inspect.signature(f)
func_args = dict(func_sig.bind_partial(*args, **kwargs).arguments)
```